### ISSUE TITLE
NVDA fires off incorrect button events 

### ISSUE THREAD

**Issue by nitronova** [nvaccess/nvda#7623]
The link below contains my expansion of the pull-out menu design originally conceived by Marcy Sutton. My desired effect was to create a navigation menu that included sub-menus. In the context of screen readers, the elements in this menu would be disabled until someone pushed the button to open the desired information.

Having tested this menu on Firefox, Chrome, and IE, the menu behaves as expected. When running a popular screen reader named NVDA, however, the wrong buttons seem to fire off. Here are the steps I take to reproduce the bug:

1. Using tab and spacebar, I open the Portfolio submenu. This highlights the Website 1 link.
2. Using shift and tab, I navigate backward to the [X] Menu ("close the main menu") button. As I do this, the portfolio button automatically closes again. Normal behavior so far.
3. Attempting to re-open the portfolio button at this point, however, results in either a) the main navigation menu button firing its event, so the menu closes, or b) the menu closes because, somehow the focus jumps to the heading (which is a link).

Codepen: https://codepen.io/nitronova/pen/qPbpMO

The main navigation button ("navButton") gets its own function called handleNavButtonClick, while submenus ("navButtons") get handleSubMenuButtonClick. Two totally separate functions. The problem is not with the functions in the script, but with how NVDA is causing the wrong functions (or event triggers) to fire.

There are no error messages; the behavior path is the problem. I have screenshots that demonstrate that the wrong events are firing when NVDA is running. https://imgur.com/a/OW1Iv

**Comment by Adriani90** [comment]
@nitronova unfortunately I cannot reproduce this issue because the codepen is not existing anymore. Could you please upload a new one? Thanks.

**Comment by nitronova** [comment]
Unfortunately, I no longer have that specific version of the code. I do not know whether or not the problem still exists. In my local copy of this project, with an updated version of NVDA, the issue does not occur.

In my original javascript code, there was some element retargeting that was going on when I would close the menu. When opening a submenu, the focus is retargeted to the first item in the list. But it doesn't retarget when you close the menu or navigate outside of it (which also closes the submenu). It was definitely a bug when I first reported it, but it doesn't happen, anymore. I wonder if it was the whole "specifying tabIndex and/or aria-hidden being applied to everything in the menu that isn't being used" part that finally made it work. I remember trying to code workarounds for months until it finally stopped bugging out. The way that menu is coded now has been functional for over a year, almost 2.

I remember that I had to be precise in where the navigation menu was arranged, but even when I put the navigation first, NVDA no longer reproduces the bug. I remember that I added a button as the first menu item and it seemed to help, but when I remove the button, NVDA doesn't produce the bug, either. So I can't seem to produce this bug, anymore.
